front/app/containers/SiteMap/index.tsx

```diff
@@ -154,216 +155,249 @@ const SiteMap = ({ projects, authUser }: Props) => {
   const hasProjectSubsection =
     archivedSection.current || draftSection.current || currentSection.current;
 
-  return (
-    <Container>
-      <SiteMapMeta />
-      {!loaded && (
-        <Loading>
-          <Spinner />
-        </Loading>
-      )}
-      {loaded && (
-        <PageContent>
-          <StyledContentContainer>
-            <QuillEditedContent>
-              <Title>
-                <FormattedMessage {...messages.siteMapTitle} />
-              </Title>
-
-              <TOC>
-                <Header>
-                  <FormattedMessage {...messages.pageContents} />
-                </Header>
-                <Ul>
-                  <li>
-                    <NavItem
-                      onMouseDown={removeFocusAfterMouseClick}
-                      onClick={scrollTo(homeSection)}
-                    >
-                      <FormattedMessage {...messages.homeSection} />
-                    </NavItem>
-                  </li>
-                  <li>
-                    <NavItem
-                      onMouseDown={removeFocusAfterMouseClick}
-                      onClick={scrollTo(userSpaceSection)}
-                    >
-                      <FormattedMessage {...messages.userSpaceSection} />
-                    </NavItem>
-                  </li>
-                  {!isNilOrError(projects) && (
+  if (!isNilOrError(pages)) {
+    const nonCustomStaticPages = pages.filter((page) => {
+      const showPageConditions: Record<TPageCode, () => boolean> = {
+        proposals: () => proposalsEnabled,
+        about: () => true,
+        faq: () => true,
+        'terms-and-conditions': () => true,
+        'privacy-policy': () => true,
+        custom: () => false,
+      };
```

Comment:
This solves the main issue, i.e. proposals should not show as part of the static pages if they are not enabled.

Reply:
Maybe a switch would be a bit simpler for me (subjective)
```ts
switch (page.attributes.code) {
  case "custom": {
    return false;
  }
  case "proposals": {
    return proposalsEnabled;
  }
  default: {
    return true;
  }
}
```

If you still prefer an object, why are the values functions and not simple booleans?

Reply:
switch looked better at first, but in fact I prefer to stick to the object, because this way we need to explicitly define the behavior for each page. If we had this before, we would not have missed it for proposals. Wdyt?